src/bin/erma/main.rs

```diff
@@ -0,0 +1,130 @@
+mod agent;
+
+use core::future::pending;
+
+use std::net::SocketAddr;
+use std::sync::Arc;
+
+use clap::Parser;
+use rand::RngCore;
+use tokio::net::{TcpListener, UdpSocket};
+
+use agent::AgentService;
+use daemonbase::error::ExitError;
+use daemonbase::logging::{self, Logger};
+use daemonbase::process::{self, Process};
+use domain::base::RelativeName;
+use domain::net::server::buf::VecBufSource;
+use domain::net::server::dgram::{self, DgramServer};
+use domain::net::server::middleware::builder::MiddlewareBuilder;
+use domain::net::server::middleware::chain::MiddlewareChain;
+use domain::net::server::middleware::processors::cookies::CookiesMiddlewareProcessor;
+use domain::net::server::stream::StreamServer;
+use domain::net::server::{stream, ConnectionConfig};
+use tracing::{error, info};
+
+//----------- Args -----------------------------------------------------------
+
+#[derive(clap::Parser)]
+pub struct Args {
+    /// Logging related settings
+    #[command(flatten)]
+    log: logging::Args,
+
+    /// Detach from the terminal
+    #[arg(short, long)]
+    detach: bool,
+
+    /// O/S process behaviour related settings
+    #[command(flatten)]
+    process: process::Args,
+
+    /// The IP address to listen on
+    #[arg(long = "addr", value_name = "LISTEN_ADDRESS", default_value = "[::]")]
+    listen_address: String,
+
+    /// The port to listen on
+    #[arg(long = "port", value_name = "LISTEN_PORT", default_value = "53")]
+    listen_port: u16,
+}
+
+//----------- init_middleware() ----------------------------------------------
+
+fn init_middleware() -> MiddlewareChain<Vec<u8>, Vec<u8>> {
+    let mut middleware = MiddlewareBuilder::<Vec<u8>, Vec<u8>>::standard();
+    let mut server_secret = [0u8; 16];
```

Comment:
If this were to be deployed on multiple servers the cookies need special attention m, e.g. to ensure they match on each server, plus there should be some mechanism to rotate them, etc, basically the guidelines in the cookie RFC should be taken into account.